src/cairoWriter/writers/cairoFunctionDefinitionWriter.ts

```diff
@@ -122,7 +122,13 @@ export class CairoFunctionDefinitionWriter extends CairoASTNodeWriter {
 
   private getReturns(node: CairoFunctionDefinition, writer: ASTWriter): string {
     if (node.kind === FunctionKind.Constructor) return '';
-    return `-> (${writer.write(node.vReturnParameters)})`;
+    const returnStr = writer.write(node.vReturnParameters);
+    if (returnStr.includes(',')) {
```

Comment:
the thing is that if the parameter is a dynamic array it will get split into 2: the len and the array, check that behavior (or any other that could be added in the future) on the function definition writer could lead to a big condition so its best to delegate that to the param list writer. The ideal solution imo would be that the param list writer returned a list of the string representation of each param, but its assumed the current returned structure in other logics, so this is the closest to that

Reply:
we're not sure though, that dynamic array returns would be split into two parameters , this examples : https://github.com/starkware-libs/cairo/blob/main/examples/fib_array.cairo tells us that we can directly an array parameter can be returned

Reply:
Ye, but that is beyond the context of this PR, perhaps when we address the representation of arrays we could solve it. @rodrigo-pino is the way of representing arrays already updated in the changes that you were working on? If so, I see no problem in change it to use the len of `vReturnParameters`

Reply:
Yes, the thing is that I was trying to not write thing with the writters that our logic not support yet, that's why I was asking if we already do that process of dynamic arrays. But ok, I will change it now anyway.